modules/database/rds/stack.py

```diff
@@ -0,0 +1,139 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import json
+from typing import Any, cast
+
+import aws_cdk as cdk
+import cdk_nag
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import aws_rds as rds
+from aws_cdk import aws_secretsmanager as sm
+from constructs import Construct, IConstruct
+
+
+def _get_hosted_rotation_for_engine(engine: str) -> sm.HostedRotation:
+    if engine == "mysql":
+        return sm.HostedRotation.mysql_single_user()
+    elif engine == "postgresql":
+        return sm.HostedRotation.postgre_sql_single_user()
+    else:
+        raise ValueError(f"Unsupported engine: {engine}")
+
+
+def _get_db_instance_engine(engine: str) -> rds.IInstanceEngine:
+    if "mysql" == engine:
+        return rds.DatabaseInstanceEngine.mysql(version=rds.MysqlEngineVersion.VER_8_0_35)
+    elif "postgresql" == engine:
+        return rds.DatabaseInstanceEngine.postgres(version=rds.PostgresEngineVersion.VER_16_1)
+    else:
+        raise ValueError(f"Unsupported engine: {engine}")
+
+
+class RDSDatabaseStack(cdk.Stack):
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        project_name: str,
+        deployment_name: str,
+        module_name: str,
+        vpc_id: str,
+        subnet_ids: list[str],
+        engine: str,
+        username: str,
+        instance_type: str,
+        removal_policy: cdk.RemovalPolicy,
+        port: int | None = None,
+        stack_description: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+
+        super().__init__(scope, id, description=stack_description, **kwargs)
+
+        self.project_name = project_name
+        self.deployment_name = deployment_name
+        self.module_name = module_name
+
+        dep_mod = f"{self.project_name}-{self.deployment_name}-{self.module_name}"
+        dep_mod = dep_mod[:64]
+        cdk.Tags.of(scope=cast(IConstruct, self)).add(key="Deployment", value=dep_mod)
+
+        # Create secret and generate password
+        self.db_credentials_secret = sm.Secret(
```

Comment:
Does this module always generate it's own PWD?  If so, that is cool.

But you need to explain this int he README...the end user needs to know if they need to provide a PWD or how they can get it...

Reply:
Yap, it generates the password along with automatic credential rotation. I'll include an explanation in the README